### PR TITLE
[Intel MKL] Compilation fixes for DNNL 1.0

### PR DESCRIPTION
**Pull request by nhasabni**
This PR makes following changes:

1. Enables MklLayoutPass for DNNL 1.0 integration
2. Makes certain changes to MKL kernels to fix compilation issues introduced by DNNL 1.0
3. Ports remaining features such as weight caching to DNNL 1.0 API
4. Adds `exponential_avg_factor` attribute to MklFusedBatchNorm ops
5. Adds BUILD support for DNNL 1.0

**NOTE**: Changes in `mkl_fused_batchnorm_op.cc` file reflect most of the changes made by PR #36767. Once this PR is merged to master, most of the changes in this file would go away.

### CODE REVIEW

**Comment by nhasabni** [comment]
Hi @penpornk, pls take a look whenever you get chance.

File: tensorflow/core/util/mkl_types.h
Comment (penpornk) [inline tensorflow/core/util/mkl_types.h]: This is redundant with line 26. Please delete one. (And please keep the one for v0 below in the same relative order.)

File: tensorflow/core/kernels/mkl_matmul_ops_common.h
Comment (penpornk) [inline tensorflow/core/kernels/mkl_matmul_ops_common.h]: Nit: can we make this weight_size?
```suggestion
    size_t weight_size = matmul_fwd_pd.get()->PRIMITIVE_DESC_WEIGHTS.get_size();
```

File: third_party/mkl_dnn/mkldnn_v1.BUILD
Comment (penpornk) [inline third_party/mkl_dnn/mkldnn_v1.BUILD]: Could you please explain we why need a seperate file? (I'm not against it. I think it's good for safety. Just curious.)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have two minor changes requested and a few comments. I haven't looked at the files that overlap with PR #36767 yet. Will do after conflicts are merged.

File: tensorflow/core/kernels/mkl_matmul_ops_common.h
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_matmul_ops_common.h]: Thanks for reviewing, Penporn! Will address it.

File: third_party/mkl_dnn/mkldnn_v1.BUILD
Comment (nhasabni) [reply third_party/mkl_dnn/mkldnn_v1.BUILD]: We thought about adding it to same mkldnn.BUILD file, but there were a lot of conditional checks that were complicating dependencies. And we would remove dnnl0.x sometime in the future. So having a separate file would make removal easy.

File: tensorflow/core/kernels/mkl_matmul_ops_common.h
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_matmul_ops_common.h]: Done. Pls take a look.

File: tensorflow/core/util/mkl_types.h
Comment (nhasabni) [reply tensorflow/core/util/mkl_types.h]: Done. Pls take a look.

File: tensorflow/core/kernels/mkl_matmul_op_fused.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_matmul_op_fused.cc]: This reformatting shouldn't happen. Please revert.

File: tensorflow/core/kernels/mkl_matmul_op_fused.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_matmul_op_fused.cc]: This reformatting shouldn't happen. Please revert.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: This is redundant to lines 29-30. Please remove.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: Thank you for the clean-up!

File: third_party/mkl_dnn/mkldnn_v1.BUILD
Comment (penpornk) [reply third_party/mkl_dnn/mkldnn_v1.BUILD]: Makes sense. Thank you!

**Review by penpornk — CHANGES_REQUESTED** [review]
There are two reformattings that shouldn't  happen. Are you using `clang-format` 8.0.1?

**Comment by nhasabni** [comment]
> There are two reformattings that shouldn't happen. Are you using `clang-format` 8.0.1?

Yes, we are using clang+llvm-8.0.1. These changes were suggested by that tool.

**Comment by penpornk** [comment]
Got it. I'll try to find the version that works some time later then T-T. Please revert by hand for now.

File: tensorflow/core/kernels/mkl_matmul_op_fused.cc
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_matmul_op_fused.cc]: Done.

File: tensorflow/core/kernels/mkl_matmul_op_fused.cc
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_matmul_op_fused.cc]: Done.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: Done.

File: .bazelrc
Comment (penpornk) [inline .bazelrc]: Now that we have time, can we separate this to another PR? Just so it's cleaner/easier for debugging.

**Review by penpornk — CHANGES_REQUESTED** [review]
One last nit, please :)

File: .bazelrc
Comment (nhasabni) [reply .bazelrc]: @penpornk should I submit separate PR now? or wait for this PR to approve and merge?

**Review by penpornk — APPROVED** [review]
Thank you for your patience! I think you can submit the other PR now. I'll make sure it's merged after this PR is merged.

**Comment by nhasabni** [comment]
@penpornk Thanks for quick review and approval. I've submitted another PR #37106 for making DNNL 1.2 default.

**Pull request merged**
